### PR TITLE
Multi-Graph-Supported-FalkorDB

### PR DESCRIPTION
**Pull request by galshubeli**
# Description

Switch graph:

Feature Introduction: Adds the switch_graph(graph_name: str) method to allow switching between graphs on the same database connection.

Why It's Needed:
Improves efficiency when working with multiple graphs under the same driver.
Removes the need to open a new connection for each graph, reducing connection overhead.

Use Case:
Ideal for applications managing several graphs simultaneously, where quick transitions between graphs are required without disconnecting or re-establishing new connections.

Goal:
Enhance performance and streamline workflows by maintaining a single connection while managing different graphs.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [x] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by logan-markewich** [comment]
@galshubeli is it possible to add a test or two that actually runs in CICD? Seems like coverage is quite low

**Comment by galshubeli** [comment]
> @galshubeli is it possible to add a test or two that actually runs in CICD? Seems like coverage is quite low

I added tests, but I need to find a way to add an environment variable. How can I do that?

**Comment by logan-markewich** [comment]
Seems like the tests you added require connecting to an actual falkordb instance

Either
- mark them as skip if the instance isn't reachable
- mock out the client interactions so that they can run in CICD
- if falkordb has an "in-instance in-memory mode" we could use that

File: llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/falkordb_property_graph.py
Comment (logan-markewich) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/falkordb_property_graph.py:1]: Should we add this to the property graph store? Just realized only the other graph store is updated

For reference, technically the property graph store is what most users would be using -- we can add to another PR though

**Review by logan-markewich — COMMENTED** [review]
Can you bump the version of the integration? (I can't actually push to the PR branch)

**Comment by galshubeli** [comment]
You're right. I'll add it to the property graph and bump the version
accordingly.

Could you assist me with the graph client? Specifically, is it possible to
use Docker in your CI? Additionally, I haven't been able to find a way to
mock the client while still testing the graph methods.

Thank you very much for your help!

Best regards,
Gal Shubeli

**Comment by galshubeli** [comment]
> Can you bump the version of the integration? (I can't actually push to the PR branch)

Is there a way to use Docker in this workflow? Mocking the client not test the connection with the graph.

@logan-markewich

**Comment by logan-markewich** [comment]
@galshubeli you can technically spin up docker yea, there's a few unit tests that do this

Check out `llama-index-integrations/vector-stores/llama-index-vector-stores-redis/tests`

But also, mocking the client is fine, assuming the mocked responses are accurate

**Review by AviAvni — APPROVED** [review]

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**